os/net/mac/csma/csma-output.c

```diff
@@ -168,11 +168,15 @@ backoff_period(void)
 static int
 send_one_packet(struct neighbor_queue *n, struct packet_queue *q)
 {
+  int expect_ack;
   int ret;
   int last_sent_ok = 0;
 
+  expect_ack = !packetbuf_holds_broadcast() && CSMA_REQUEST_ACK;
+
   packetbuf_set_addr(PACKETBUF_ADDR_SENDER, &linkaddr_node_addr);
-  packetbuf_set_attr(PACKETBUF_ATTR_MAC_ACK, 1);
+
+  packetbuf_set_attr(PACKETBUF_ATTR_MAC_ACK, expect_ack);
```

Comment:
ok we had a bug there, setting `PACKETBUF_ATTR_MAC_ACK` even for broadcast, but somehow didn't have any effect, as the framer didn't look at this attribute for broadcast, and left the ACK Requested frame field unset?